File: 4 graphs_algorithms/2_bfs_dfs/bfs.cs
using System;
using System.Collections.Generic;

class Program
{
    static void Main(string[] args)
    {
        Dictionary<char, List<char>> graph = new Dictionary<char, List<char>>
        {
            { 'A', new List<char> { 'B', 'C' } },
            { 'B', new List<char> { 'A', 'D', 'E' } },
            { 'C', new List<char> { 'A', 'F' } },
            { 'D', new List<char> { 'B' } },
            { 'E', new List<char> { 'B', 'F' } },
            { 'F', new List<char> { 'C', 'E' } }
        };

        BFS(graph, 'A');
    }

    static void BFS(Dictionary<char, List<char>> graph, char start)
    {
        HashSet<char> visited = new HashSet<char>(); // Множина для відвіданих вершин
        Queue<char> queue = new Queue<char>();      // Черга для обробки вершин

        queue.Enqueue(start);                       // Додаємо початкову вершину у чергу

        while (queue.Count > 0)
        {
            char vertex = queue.Dequeue();          // Беремо вершину з початку черги
            if (!visited.Contains(vertex))
            {
                Console.WriteLine(vertex);         // Обробляємо вершину
                visited.Add(vertex);               // Позначаємо вершину як відвідану
                foreach (char neighbor in graph[vertex])
                {
                    if (!visited.Contains(neighbor))
                    {
                        queue.Enqueue(neighbor);   // Додаємо всі непомічені сусідні вершини у чергу
                    }
                }
            }
        }
    }
}

File: college/day-department/algorithms/5 graphs_algorithms/2_bfs_dfs/bfs.py
from collections import deque

def bfs(graph, start):
    visited = set()            # Множина для відвіданих вершин
    queue = deque([start])     # Черга для обробки вершин

    while queue:
        vertex = queue.popleft()    # Беремо вершину з початку черги
        if vertex not in visited:
            print(vertex)          # Обробляємо вершину
            visited.add(vertex)    # Позначаємо вершину як відвідану
            for neighbor in graph[vertex]:
                if neighbor not in visited:
                    queue.append(neighbor)   # Додаємо всі непомічені сусідні вершини до черги

# Приклад графа у форматі списку сусідів
graph = {
    'A': ['B', 'C'],
    'B': ['A', 'D', 'E'],
    'C': ['A', 'F'],
    'D': ['B'],
    'E': ['B', 'F'],
    'F': ['C', 'E']
}

# Запуск BFS з початковою вершиною 'A'
bfs(graph, 'A')

